### ISSUE TITLE
How can we get logits of vocab in BB3 ( #4967 )

### ISSUE THREAD

**Issue by SarikGhazarian** [facebookresearch/ParlAI#5054]
Hi,

I am using BB3 to complete dialogue context in non-interactive mode (thanks for your guidance in #4967). In each timestep of generation I need to get access to logits of tokens. I added "text_candidate_scores": True to Message but I was not able to get logits.

act_0 = Message({"text": query, "episode_done": True, "text_candidate_scores": True})

**Comment by mojtaba-komeili** [comment]
I don't think ParlAI adds the logic of the output message. We are using those value for calculating PPL and the overall loss. Your best solution might be adding your custom code to add those to the metrics. Look into [this function](https://github.com/facebookresearch/ParlAI/blob/ec2132811f6d51e63fc645df3cf9ff69b9bd3b37/parlai/core/torch_generator_agent.py#L732) for more information.  You may add the custom code by overriding [this function](https://github.com/facebookresearch/ParlAI/blob/ec2132811f6d51e63fc645df3cf9ff69b9bd3b37/parlai/core/torch_generator_agent.py#L726) in your extended agent. Feel free to submit your diff to our repository.
